### PR TITLE
Pathfinder

### PR DESCRIPTION
**Pull request by bhollis**
I took a stab at showing the Pathfinders in the Progress page. They're similar to the Seasonal Challenges, with a few differences:

1. Round icons
2. It's arranged in an inverted triangle, like in game, but rotated. It should be generally easy to figure out what the paths are between them.
3. On mobile, the text goes away so we can prioritize the triangle over showing all the details.
4. Completed records are still shown, just with a green border.

Screenshot 2024-06-18 at 12 50 44 AM
Screenshot 2024-06-18 at 12 50 56 AM

Definitely take this for a spin through the PR deployment link. LMK if you have any more design suggestions.

### CODE REVIEW

**Comment by bhollis** [comment]
Note: I considered centering the icons, but I don't think it improved anything:

Screenshot 2024-06-18 at 12 59 54 AM

File: src/app/progress/milestone-items.ts [hunk 1]
```diff
@@ -385,7 +385,9 @@ export function recordToPursuitItem(
   };
 
   if (record.recordDef.rewardItems) {
-    dimItem.pursuit.rewards = record.recordDef.rewardItems;
+    dimItem.pursuit.rewards = record.recordDef.rewardItems.filter(
+      (_r, i) => record.recordComponent.rewardVisibilty?.[i] ?? true,
+    );
```
Comment (bhollis) [inline src/app/progress/milestone-items.ts:390]: This is fun, turns out there was a way to tell which rewards are actually visible!

**Comment by delphiactual** [comment]
I want to point out that this is a valid path
image

**Review by lowPolySkeleton — COMMENTED** [review]
Won't be near a pc for good while but I think this is a solid start, maybe we restrict them all to the same width so we can get that pyramid alignment. Or restrict their height if we want to go with the horizontal layout. I think we just need to make them conform on whatever dimension we want the pyramid to go. And yeah adding boxes while typically I'd be against for this I think would again help with that alignment aspect that is important for pathfinder

**Review by delphiactual — APPROVED** [review]

**Pull request merged**